tests/tracking/test_fluent.py

```diff
@@ -681,3 +682,42 @@ def test_delete_tag():
     with pytest.raises(MlflowException):
         mlflow.delete_tag("b")
     mlflow.end_run()
+
+
+@pytest.mark.large
+def test_universal_autolog_calls_specific_autologs_correctly(mocker):
+    autolog_integrations = mlflow.tracking.fluent.AUTOLOG_INTEGRATIONS
+    integrations_with_config = ["xgboost", "lightgbm", "sklearn"]
+
+    for integration_name in autolog_integrations.keys():
+        # modify the __signature__ of the mock to contain the needed parameters
+        args = (
+            {"log_input_example": bool, "log_model_signature": bool}
+            if integration_name in integrations_with_config
+            else {}
+        )
+        params = [
+            inspect.Parameter(param, inspect.Parameter.POSITIONAL_OR_KEYWORD, annotation=type_)
+            for param, type_ in args.items()
+        ]
+
+        autolog_fn_spy = mocker.spy(
+            getattr(mlflow, autolog_integrations[integration_name]), "autolog"
+        )
+        autolog_fn_spy.__signature__ = inspect.Signature(params)
+
+        setattr(getattr(mlflow, autolog_integrations[integration_name]), "autolog", autolog_fn_spy)
+        autolog_fn = getattr(getattr(mlflow, autolog_integrations[integration_name]), "autolog")
+        autolog_fn.assert_not_called()
+
+    mlflow.autolog(log_input_example=True, log_model_signature=True)
```

Comment:
I think we should be performing `autolog_fn.assert_not_called()` after `mlflow.autolog()` is invoked; this should hold for all libraries except PySpark (for which, as discussed in a comment above, we want to invoke autologging as soon as `mlflow.autolog()` is called).

Otherwise, it's possible that `mlflow.autolog()` calls all of these autologging functions directly and our test doesn't catch it.

Reply:
if we take pyspark as a separate case as described, then yeah we can put `assert_not_called` back as part of this test. Previously, I had to remove it specifically because of pyspark